x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/organization_sources.tsx

```diff
@@ -36,33 +35,37 @@ export const OrganizationSources: React.FC = () => {
 
   const { dataLoading, contentSources } = useValues(SourcesLogic);
 
-  if (dataLoading) return <Loading />;
-
-  if (contentSources.length === 0) return <Redirect to={getSourcesPath(ADD_SOURCE_PATH, true)} />;
```

Comment:
I like the idea of putting a Redirect in `emptyState` prop, but it doesn't work well:

https://user-images.githubusercontent.com/11838280/122589107-3cb15c80-d036-11eb-8b5e-48fa0c392417.mp4

Reply:
A similar thing happens on the Overview page. It might not be related to this PR, but posting here, because it could require the same fix.

https://user-images.githubusercontent.com/11838280/122589950-438c9f00-d037-11eb-8949-01841d13c106.mp4

Reply:
Gotcha, can I clarify your preferred UX? Would you rather the page header not appear at all until dataLoading is done? If so I can stick a `dataLoading ?` conditional in front of that prop.

Reply:
Also TIL GitHub lets you attach mp4s!!!!

Reply:
I think I would prefer the UX work like this (in the order of preference):
1. Show the correct header, description, and loading indicator while loading (I think it might be possible on Overview page)
2. Show only loading indicator until we know for sure what header and description we need to show (should probably be used on Sources page)

Reply:
> Show the correct header, description, and loading indicator while loading (I think it might be possible on Overview page)

I don't _think_ this is possible on the Overview page because it's all the required data to check if we're on the onboarding state is initialized in OverviewLogic. Unless the below data is initialized at the app level before the Overview page loads?

https://github.com/elastic/kibana/blob/1627240ff8b8f82e41de5169b5d4dedbb88447c1/x-pack/plugins/enterprise_search/public/applications/workplace_search/views/overview/overview.tsx#L48

Reply:
Here's the updated UX with `pageHeader` no longer rendering before data is done loading which conditionally affects the title/redirects:

![ux](https://user-images.githubusercontent.com/549407/122595661-a4a77900-d01d-11eb-9b9d-be68d581453f.gif)

So effectively it reverts Overview and Sources to the pre-KibanaPageTemplate UX where the entire page is blank/loading instead of immediately loading in a page title. I contrasted it with other pages that have non-conditional `pageHeader`s at the end of the gif.

IMO: pages w/o a header that load in asap feels a _little_ less "snappy", but I agree with you it's a little more confusing that the header jumps around after load. However, another consideration is that you're trading a 'slightly' worse UX for new users vs a slightly nicer UX for established users 🤪 If we really really want the best of both worlds, I'd suggest a way to load in an `emptySources` or `isOnboarding` state at the app-wide level before any pages load - that way you're not waiting for individual page `dataLoading` states before changing the page, you can just show the page you want immediately.

Reply:
Ah, I see. Let's go with option 2 in both cases then as you suggested?

Reply:
Addressed in https://github.com/elastic/kibana/pull/102592/commits/8bea24ec554be6cefde478e6590391c8d0389e99!

Reply:
I missed your last comment! 
I like your suggestion of loading `emptySources` and `isOnboarding` state at the app-wide level. We'll do it as part of polish (already in the polish issue).